lib/dialogs.dart

```diff
@@ -109,6 +112,11 @@ class SharingDialog extends DDialog {
     _gistUrl.onClick.listen((_) => _gistUrl.selectAll());
   }
 
+  void updateSummary(String summary) {
```

Comment:
Let's call this something else, to indicate that it is also makes the dialog visible. `showWithSummary()`? `showSharingDialog(String summary)`?